packages/gatsby-source-contentful/src/__tests__/hooks.js

```diff
@@ -0,0 +1,71 @@
+import { useContentfulImage } from "../hooks"
+
+describe(`useContentfulImage`, () => {
+  const consoleWarnSpy = jest.spyOn(console, `warn`)
+
+  const image = { url: `//images.ctfassets.net/foo/bar/baz/image.jpg` }
+
+  beforeEach(() => {
+    consoleWarnSpy.mockClear()
+  })
+
+  test(`constrained: width and aspectRatio`, () => {
+    const result = useContentfulImage({
+      image,
+      width: 600,
+      aspectRatio: 1.778,
+    })
+    expect(result).toMatchSnapshot()
```

Comment:
It would be good to expect some values here as well to make the intent of the tests clearer